packages/react/src/executors/module-federation-dev-server/module-federation-dev-server.impl.ts

```diff
@@ -37,89 +41,71 @@ function getBuildOptions(buildTarget: string, context: ExecutorContext) {
   };
 }
 
-export default async function* moduleFederationDevServer(
-  options: ModuleFederationDevServerOptions,
-  context: ExecutorContext
-): AsyncIterableIterator<{ success: boolean; baseUrl?: string }> {
-  // Force Node to resolve to look for the nx binary that is inside node_modules
-  const nxBin = require.resolve('nx/bin/nx');
-  const currIter = options.static
-    ? fileServerExecutor(
-        {
-          ...options,
-          parallel: false,
-          withDeps: false,
-          spa: false,
-          cors: true,
-        },
-        context
-      )
-    : devServerExecutor(options, context);
-
-  const p = context.projectsConfigurations.projects[context.projectName];
-  const buildOptions = getBuildOptions(options.buildTarget, context);
+function startStaticRemotesFileServer(
+  remotes: {
+    remotePorts: any[];
+    staticRemotes: string[];
+    devRemotes: string[];
+  },
+  context: ExecutorContext,
+  options: ModuleFederationDevServerOptions
+) {
+  let shouldMoveToCommonLocation = false;
+  let commonOutputDirectory: string;
+  for (const app of remotes.staticRemotes) {
+    const outputPath =
+      context.projectGraph.nodes[app].data.targets['build'].options.outputPath;
+    const directoryOfOutputPath = dirname(outputPath);
 
-  if (!options.isInitialHost) {
-    return yield* currIter;
+    if (!commonOutputDirectory) {
+      commonOutputDirectory = directoryOfOutputPath;
+    } else if (commonOutputDirectory !== directoryOfOutputPath) {
+      shouldMoveToCommonLocation = true;
+    }
   }
 
-  const moduleFederationConfig = getModuleFederationConfig(
-    buildOptions.tsConfig,
-    context.root,
-    p.root,
-    'react'
-  );
+  if (shouldMoveToCommonLocation) {
+    commonOutputDirectory = join(workspaceRoot, 'tmp/static-remotes');
+    for (const app of remotes.staticRemotes) {
+      const outputPath =
+        context.projectGraph.nodes[app].data.targets['build'].options
+          .outputPath;
+      cpSync(outputPath, commonOutputDirectory, {
+        force: true,
+        recursive: true,
+      });
+    }
+  }
 
-  const remotes = getRemotes(
-    options.devRemotes,
-    options.skipRemotes,
-    moduleFederationConfig,
+  const staticRemotesIter = fileServerExecutor(
     {
-      projectName: context.projectName,
-      projectGraph: context.projectGraph,
-      root: context.root,
-    }
+      skipInitialRun: true,
```

Comment:
Instead of a new option, can we make `buildTarget` optional in `file-server`, and skip watching/running build in that case?

1. If `buildTarget` is not present, `staticFilePath` _must_ be present, or else throw error
2. If `watch` is used but `buildTarget` is not present, then throw error

This makes `file-server` able to run any path, detached from any build process.

Reply:
We can, I was more apprehensive about making a large change to file-server as it's used for static serve of apps for cypress e2es etc too and it seemed like a high risk change to make.

But if this is preferable, I can do that.